### PR TITLE
Get the ball rolling with the first mTLS IT test case..

### PR DESCRIPTION
**Pull request by k-wall**
First IT test case testing the happy path.  I think we'd want to cover a couple more cases, but I wanted to demonstrate the way the ITs work.

### CODE REVIEW

File: kroxylicious-api/src/main/java/io/kroxylicious/proxy/config/tls/Tls.java [hunk 1]
```diff
@@ -45,6 +36,15 @@ public Tls(KeyProvider key, TrustProvider trust, TlsClientAuth clientAuth) {
         validateClientAuth();
     }
 
+    /**
+     * @deprecated use the all args constructor {@see io.kroxylicious.proxy.config.tls.Tls#Tls(io.kroxylicious.proxy.config.tls.KeyProvider, io.kroxylicious.proxy.config.tls.TrustProvider, io.kroxylicious.proxy.config.tls.TlsClientAuth)}
+     */
+    // This is required for API backwards compatability
```
Comment (k-wall) [inline kroxylicious-api/src/main/java/io/kroxylicious/proxy/config/tls/Tls.java:42]: Weird.. Sundrio seems to care about constructor order.  If we have the ctors the other way around, the builder omits the new field.

**Comment by alanrobinson-dwp** [comment]
Thanks @k-wall just seen this

File: kroxylicious-integration-test-support/src/main/java/io/kroxylicious/proxy/config/BuilderConfig.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
         "io.kroxylicious.proxy.config.secret.FilePassword",
         "io.kroxylicious.proxy.config.secret.InlinePassword",
         "io.kroxylicious.proxy.config.tls.Tls",
+        "io.kroxylicious.proxy.config.tls.TlsClientAuth",
```
[the remark this answers is not in the document]
Comment (SamBarker) [reply kroxylicious-integration-test-support/src/main/java/io/kroxylicious/proxy/config/BuilderConfig.java:24]: Ahhh I went looking for maven config and didn't find it before I ran out is day.

[the remark this answers is not in the document]
Comment (SamBarker) [reply kroxylicious-integration-test-support/src/main/java/io/kroxylicious/proxy/config/BuilderConfig.java:24]: Changing the ctor order is enough to generate the `withClientAuth` methods.

Which messes with my head.

[the remark this answers is not in the document]
Comment (SamBarker) [reply kroxylicious-integration-test-support/src/main/java/io/kroxylicious/proxy/config/BuilderConfig.java:24]: see https://github.com/alanrobinson-dwp/kroxylicious/pull/3

**Review by alanrobinson-dwp — APPROVED** [review]

**Pull request merged**